packages/btc/src/query/source.ts

```diff
@@ -158,19 +160,18 @@ export class DataSource {
     };
   }
 
-  // Get recommended fee rates from mempool.space.
-  // From fastest to slowest: fastestFee > halfHourFee > economyFee > hourFee > minimumFee
-  async getRecommendedFeeRates(): Promise<FeesRecommended> {
-    try {
-      return await this.mempool.bitcoin.fees.getFeesRecommended();
-    } catch (err: any) {
-      throw TxBuildError.withComment(ErrorCodes.MEMPOOL_API_RESPONSE_ERROR, err.message ?? JSON.stringify(err));
-    }
+  /**
+   * Get recommended fee rates from mempool.space.
+   * From fastest to slowest: fastestFee > halfHourFee > economyFee > hourFee > minimumFee
+   * @deprecated Use BtcAssetsApi.getBtcRecommendedFeeRates() instead.
+   */
+  async getRecommendedFeeRates(): Promise<BtcApiRecommendedFeeRates> {
+    return await this.service.getBtcRecommendedFeeRates();
   }
 
   // Get the recommended average fee rate.
   async getAverageFeeRate(): Promise<number> {
-    const fees = await this.getRecommendedFeeRates();
+    const fees = await this.service.getBtcRecommendedFeeRates();
     return fees.halfHourFee;
   }
 
```

Comment:
Maybe we just need `getRecommendedFeeRates`, `getAverageFeeRate` looks like it just goes to one of the fields. Are there compatibility considerations here?

Reply:
Yes, personally I would like to remove both `DataSource.getRecommendedFeeRates` and `DataSource.getAverageFeeRate`, but I didn't do it due to compatibility concerns. The PR #98 was stuck for the same reason. I think if JoyID is not using them directly, we can remove them with a `BREAKING CHANGE` mark in the PR. 

@yuche Do you think we can remove these APIs?